python-backend/app/api/parties/party/resources/party_advance_search_resource.py

```diff
@@ -0,0 +1,98 @@
+    parser.add_argument('first_name', type=str,
+        help='First name of the party, if the party is a person.')
+    parser.add_argument('last_name', type=str,
+                        help='Last name of the party, if the party is a person.')
+    parser.add_argument('party_name', type=str,
+        help='Last name of the party (Person), or the Organization name (Organization).')
+    parser.add_argument('phone_no', type=str,
+        help='The phone number of the party. Ex: 123-123-1234')
+    parser.add_argument('phone_ext', type=str, help='The extension of the phone number. Ex: 1234')
+    parser.add_argument('email', type=str, help='The email of the party.')
+    parser.add_argument('type', type=str, help='The type of the party. Ex: PER')
+    parser.add_argument('page', type=int, help='The page of the results.')
+    parser.add_argument('per_page', type=int, help='The number of parties per page.')
+    parser.add_argument('role', type=str, help='The role of a user. Ex: MMG for Mine Manager')
+    PARTY_LIST_RESULT_LIMIT = 25
+
+
+    @api.doc(
+        params={
+            'first_name': 'First name of party or contact',
+            'party_name': 'Last name or party name of person or organisation',
+            'phone_no': 'phone number',
+            'email': 'email of person or organisation',
+            'type': 'A person (PER) or organisation (ORG)',
+            'role': 'A comma separated list of roles to be filtered by'
+        })
+    @requires_any_of([MINE_VIEW, MINESPACE_PROPONENT])
+    def get(self):
+        paginated_parties, pagination_details = self.apply_filter_and_search(request.args)
+        if not paginated_parties:
+            self.raise_error(
+                404,
+                'No parties found'
+            ), 404
+        parties = paginated_parties.all()
+        return {
+            'parties': list(map(lambda x: x.json(relationships=['mine_party_appt']), parties)),
+            'current_page': pagination_details.page_number,
+            'total_pages': pagination_details.num_pages,
+            'items_per_page': pagination_details.page_size,
+            'total': pagination_details.total_results,
+        }
+
+    def apply_filter_and_search(self, args):
+        # Handle ListView request
+        items_per_page = args.get('per_page', 25, type=int)
+        page = args.get('page', 1, type=int)
+        # parse the filter terms
+        first_name_filter_term = args.get('first_name', None, type=str)
+        last_name_filter_term = args.get('last_name', None, type=str)
+        party_name_filter_term = args.get('party_name', None, type=str)
+        type_filter_term = args.get('type', None, type=str)
+        role_filter_term = args.get('role', None, type=str)
+        email_filter_term = args.get('email', None, type=str)
+        phone_filter_term = args.get('phone_no', None, type=str)
+
+        conditions = []
+        if first_name_filter_term and len(first_name_filter_term.strip()) > 0:
```

Comment:
you can add `trim=True` to the add_argument() calls above and it come out stripped.

Reply:
that is a useful trick

Reply:
done